packages/react-sdk/src/useObservable.ts

```diff
@@ -83,30 +97,31 @@ export function useObservable<T>(
     }, [opts, observable, onSubscribe])
 
     const data = useMemo(() => {
-        if (!value) {
+        if (isPersisted(value)) {
+            const { data, status } = value
             return {
-                data: undefined,
+                data: data,
+                error: status === 'error' ? value.error : undefined,
+                status,
+                isLoading: status === 'loading',
+                isError: status === 'error',
+                isSaving: status === 'saving',
+                isLoaded: status === 'loaded',
+                isSaved: status === 'saved',
+            }
+        } else {
+            return {
+                data: value,
                 error: undefined,
-                status: 'loading',
-                isLoading: true,
+                status: 'loaded',
+                isLoading: false,
                 isError: false,
                 isSaving: false,
-                isLoaded: false,
+                isLoaded: true,
                 isSaved: false,
             }
         }
-        const { data, status } = value
-        return {
-            data,
-            error: status === 'error' ? value.error : undefined,
-            status,
-            isLoading: status === 'loading',
-            isError: status === 'error',
-            isSaving: status === 'saving',
-            isLoaded: status === 'loaded',
-            isSaved: status === 'saved',
-        }
-    }, [value]) satisfies ObservableReturn<T>
+    }, [value]) as ObservableValue<T>
```

Comment:
Unfortunately, I hit some issues with the TS compiler to understand that
```ts
if (isPersisted(value) {
   // means that T extends PersistedModel<T>
} else {
  // means that T dont extend PersistedModel<T>
}
```
that's why im using `as` here. Would love to remove, maybe TS 5.5 is smart enough?